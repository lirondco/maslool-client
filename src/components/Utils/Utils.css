.Input {
  border-radius: 15px;
  background-color: #fffcf5;
  color: #374355;
  border-style: solid;
  border-width: 2px;
  width: max-content;
}

.Textarea {
  padding: 2%;
  background-color: white;
  border: #524c57 solid 1px;
  border-radius: 10px;
  width: 96%;
}

.Required {
  color: red;
}

.Button {
  border-radius: 8px;
  background-color: #524c57;
  color: #fff;
  font-weight: bolder;
  font-size: 12pt;
  padding: 4px 20px;
}

.Button:disabled {
  background-color: gray;
}

.Button:active {
  background-color: brown;
}
