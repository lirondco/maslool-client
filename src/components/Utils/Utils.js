import React from 'react'
import cx from 'classnames'
import './Utils.css'

export function Label({ className, ...props }) {
  return (
    <label className={cx('Label', className)} {...props} />
  )
}

export const Input =  React.forwardRef(({ className, ...props }, ref) => {
  return (
    <input className={cx('Input', className)} type='text' ref={ref} {...props} />
  )
})

export function Required({ className, ...props }) {
  return (
    <span className={cx('Required', className)} {...props}>
      &#42;
    </span>
  )
}

export function Textarea({ className, ...props }) {
  return (
    <textarea className={cx('Textarea', className)} {...props} />
  )
}

export const Button = React.forwardRef(({ className, ...props }, ref) => {
    return (
      <button className={cx('Button', className)} ref={ref} {...props} />
    )
  })
  