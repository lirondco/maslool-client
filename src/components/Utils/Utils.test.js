import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { Button, Input, Textarea, Required, Label, NiceDate } from './Utils'

describe(`Utils components`, () => {
    const props = {
        className: 'test-class-name',
        children: <p>test children</p>,
        'data-other': 'test-other-prop'
    }

    const testDate = '2021-03-11T06:47:06.477Z'

    it('renders a date when NiceDate is called', () => {
        const wrapper = shallow(<NiceDate date={testDate} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('reders a button when called', () => {
        const wrapper = shallow(<Button {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('button works when clicked', () => {
        const wrapper = shallow(<Button {...props} />)
        wrapper.find('button').simulate('click', { preventDefault() {}})
    })

    it('Input renders when called', () => {
        const wrapper = shallow(<Input {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('Textarea renders when called', () => {
        const wrapper = shallow(<Textarea {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('Required renders when called', () => {
        const wrapper = shallow(<Required {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('Label renders when called', () => {
        const wrapper = shallow(<Label {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})