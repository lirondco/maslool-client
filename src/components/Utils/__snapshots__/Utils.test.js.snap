// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utils components Input renders when called 1`] = `
<input
  className="Input test-class-name"
  data-other="test-other-prop"
  type="text"
>
  <p>
    test children
  </p>
</input>
`;

exports[`Utils components Label renders when called 1`] = `
<label
  className="Label test-class-name"
  data-other="test-other-prop"
>
  <p>
    test children
  </p>
</label>
`;

exports[`Utils components Required renders when called 1`] = `
<span
  className="Required test-class-name"
  data-other="test-other-prop"
>
  *
</span>
`;

exports[`Utils components Textarea renders when called 1`] = `
<textarea
  className="Textarea test-class-name"
  data-other="test-other-prop"
>
  <p>
    test children
  </p>
</textarea>
`;

exports[`Utils components reders a button when called 1`] = `
<button
  className="Button test-class-name"
  data-other="test-other-prop"
>
  <p>
    test children
  </p>
</button>
`;

exports[`Utils components renders a date when NiceDate is called 1`] = `"10 Mar 2021"`;
